code/modules/integrated_electronics/subtypes/manipulation.dm

```diff
@@ -203,4 +202,171 @@
 /obj/item/integrated_circuit/manipulation/grenade/frag
 	pre_attached_grenade_type = /obj/item/weapon/grenade/explosive
 	origin_tech = list(TECH_ENGINEERING = 3, TECH_DATA = 3, TECH_COMBAT = 10)
-	spawn_flags = null			// Used for world initializing, see the #defines above.
\ No newline at end of file
+	spawn_flags = null			// Used for world initializing, see the #defines above.
+
+/obj/item/integrated_circuit/manipulation/shocker
+	name = "shocker circuit"
+	desc = "Used to shock adjastent creatures with electricity."
+	icon_state = "shocker"
+	extended_desc = "The circuit accepts a reference to creature,who needs to be shocked. It can shock target on adjastent tiles.."
+	w_class = ITEMSIZE_TINY
+	complexity = 10
+	inputs = list("target" = IC_PINTYPE_REF)
+	outputs = list()
+	activators = list("shock" = IC_PINTYPE_PULSE_IN)
+	spawn_flags = IC_SPAWN_RESEARCH
+	power_draw_per_use = 50
+
+/obj/item/integrated_circuit/manipulation/shocker/do_work()
+	..()
+	var/turf/T = get_turf(src)
+	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /mob/living)
+	if(!istype(AM,/mob/living)) //Invalid input
+		return
+	var/mob/living/M = AM
+	if(!M.Adjacent(T))
+		return //Can't reach
+	M << "<span class='danger'>You feel a sharp shock!</span>"
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(3, 1, M)
+	s.start()
+	M.Weaken(10)
+
+/obj/item/integrated_circuit/manipulation/grabber
+	name = "grabber"
+	desc = "circuit with it's own inventory for small/medium items ,used to grab and store things."
+	icon_state = "grabber"
+	extended_desc = "The circuit accepts a reference to thing to be grabbed. It can store up to 10 things.Modes:1 for grab. 0 for eject the first thing. -1 for eject all"
```

Comment:
Need a space between ``things.`` and ``Modes:1``, so it looks like ``things. Modes:1``.  Also need a period at the end.